### DISCUSSION TITLE
Does Sirius support a grid diagram?

### DISCUSSION THREAD

**Discussion by Cirtl** [eclipse-sirius/sirius-desktop#445] [Q&A] [answered]
Hello everybody, what I asked is that while creating nodes in diagram, each node must be located in a grid which is pre-defined (or in other way) in the diagram, and the node can get its coordinates of the grid where it is located.
Does Sirius support some features like this?

**Comment by pcdavid** [comment, marked as answer]
Hi,

I'm not sure this is what you're asking for, but there is support for enabling a "snap to grid" behavior.
Click on the diagram's background and in the Properties view this can be configured in the "Rulers & Grid" tab:

![image](https://github.com/user-attachments/assets/222e780c-bbd6-4b43-a2e8-35d41470c38f)

If your question is more "can I enforce the position where a node should be displayed depending on some properties in my semantic model", then no. For normal Sirius diagrams the position (and size) of a node on the canvas (or relative position inside its parent container) are considered purely graphical information with no relation to the semantic data. The only exception is for sequence diagrams which *do* have some very specific rules and constraints enforced between the relative position of elements on the diagram and the underlying semantic model.

**Reply by Cirtl** [reply]
Thank you for your answer, it's very helpful!
